packages/nx/src/command-line/run-many.ts

```diff
@@ -40,16 +57,70 @@ function projectsToRun(
 ): ProjectGraphProjectNode[] {
   const allProjects = Object.values(projectGraph.nodes);
   const excludedProjects = new Set(nxArgs.exclude ?? []);
+
   if (nxArgs.all) {
     return runnableForTarget(allProjects, nxArgs.target).filter(
       (proj) => !excludedProjects.has(proj.name)
     );
   }
+
   checkForInvalidProjects(nxArgs, allProjects);
-  let selectedProjects = nxArgs.projects.map((name) =>
-    allProjects.find((project) => project.name === name)
+
+  const groupedProjectMatchingPatterns = groupBy(
+    nxArgs.projects,
+    groupPatterns
+  ) as Record<ProjectMatchingType, string[]>;
+
+  const {
+    globPatterns = [],
+    precisePatterns = [],
+    // exclude patterns are shared in apps and libs globbing
+    excludePatterns = [],
+  } = groupedProjectMatchingPatterns;
+
+  const ignorePatternsMatching = excludePatterns.map((p) =>
+    p.startsWith('!') ? p.slice(1, p.length) : p
   );
-  return runnableForTarget(selectedProjects, nxArgs.target, true).filter(
+
+  // skip when no valid glob patterns provided
+  const shouldGlobProjects = Boolean(globPatterns.length);
+
+  // here we assume that the name of projects in the two directories are not duplicated
+  const { appsDir, libsDir } = workspaceLayout();
+
+  // glob packages in appsDir
+  const globbedAppProjects = shouldGlobProjects
+    ? globPatterns
+        .map((pattern) =>
+          globSync(pattern, {
+            cwd: path.resolve(appsDir),
+            ignore: ignorePatternsMatching,
+          })
+        )
+        .flat()
+    : [];
+
+  // glob packages in libsDir
+  const globbedLibProjects = shouldGlobProjects
+    ? globPatterns
+        .map((pattern) =>
+          globSync(pattern, {
+            cwd: path.resolve(libsDir),
+            ignore: ignorePatternsMatching,
+          })
+        )
+        .flat()
+    : [];
+
+  const matchedProjectNodeList = allProjects.filter((project) =>
```

Comment:
I think this should be much simper. 

Something along the lines of:

```
import { match } from 'minimatch'; // already a dependency

const machingProjects = match(allProjectNames, pattern);
```

It's not actually that simple since we want to also handle comma separated patterns. I don't think there should be a restriction on the pattern..

Could you please revisit this and make it simpler?

Reply:
I'll try to simplify this part of the logic